src/components/shapeComponents/setDefinition/EntityAutocomplete.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <AutoComplete :multiple="true" v-model="selectedEntity" :suggestions="suggestions" @complete="searchEntity($event)" @item-select="handleChange">
+    <template #item="slotProps"> {{ slotProps.item.name }} - {{ slotProps.item["@id"] }} </template>
+    <template #chip="slotProps">
+      <div v-tooltip.right="slotProps.value['@id']">{{ slotProps.value.name }}</div>
+    </template>
+  </AutoComplete>
+</template>
+
+<script setup lang="ts">
+import { QueryObject, SearchRequest, SetQueryObject, TTAlias, TTIriRef } from "im-library/dist/types/interfaces/Interfaces";
+import { onMounted, PropType, Ref, ref, watch } from "vue";
+import { Services, Enums, Helpers, Config } from "im-library";
+import axios from "axios";
+import _ from "lodash";
+const { EntityService } = Services;
+const { isObject, isObjectHasKeys, isArrayHasLength } = Helpers.DataTypeCheckers;
+
+const props = defineProps({
+  ttAlias: { type: Object as PropType<TTAlias>, required: true },
+  parentClauseIri: { type: String, required: false },
+  getSuggestionsMethod: { type: Function, required: false }
+});
+
+const suggestions = ref();
+const entityService = new EntityService(axios);
+const abortController = ref(new AbortController());
+const selectedEntity: Ref<TTIriRef[]> = ref([] as TTIriRef[]);
+
+onMounted(() => {
+  if (isObjectHasKeys(props.ttAlias, ["@id", "name"])) selectedEntity.value = [{ "@id": props.ttAlias["@id"], name: props.ttAlias.name }];
+});
+
+function handleChange(event: any) {
+  selectedEntity.value = [event.value];
+  props.ttAlias["@id"] = event.value["@id"];
```

Comment:
This is mutating a prop, which is a bad practice. Surprised it's working as vue usually blocks this.